### PR TITLE
Product page implemented

### PR DESCRIPTION
**Pull request by Sharookhan**

### CODE REVIEW

File: web/pages/product.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  }, [id]) // Include id in the dependency array to fetch new product when id changes
+
+  return (
+    <div>
+      {product ? (
+        <>
+          <ProductContainer product={product} />
+        </>
+      ) : (
+        <p>Loading product...</p>
```
Comment (georgevarghese185) [inline web/pages/product.jsx:33]: I believe we used a skeleton (bootstrap placeholder) in v1 for when the product is loading. Can we re-implement that? When product is null, render the skeleton/placeholder elements. You might have to implement it inside each individual component though.. Maybe you could code each inner component to either render data if product is non null or render a placeholder if product is null.

File: web/components/product/ReviewSection.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+const ReviewsSection = ({ reviews }) => {
+  if (!reviews || reviews.length === 0) {
+    console.log(reviews)
```
Comment (georgevarghese185) [inline web/components/product/ReviewSection.jsx:3]: Let's remove this console log

**Review by georgevarghese185 — CHANGES_REQUESTED** [review]
Rest of the code looks great! Love the structure. See if rendering bootstrap placeholders when loading is possible. or if there's some complication we can re-evaluate its need and I'll approve this PR

File: web/pages/product.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import { useParams } from 'react-router-dom'
+
+const Product = () => {
+  const [product, setProduct] = useState(null)
+  const { id } = useParams()
+
+  useEffect(() => {
+    async function fetchProduct() {
+      try {
+        const response = await axios.get(
```
Comment (georgevarghese185) [inline web/pages/product.jsx:13]: we already have code to fetch product details in product service. Can we copy it from v1 into here and re-use it?

File: package.json [hunk 4]
```diff
@@ -33,6 +33,7 @@
     "vite": "^5.2.0"
   },
   "dependencies": {
+    "axios": "^1.7.2",
```
Comment (georgevarghese185) [inline package.json:36]: If we re-use the existing product service from v1 we won't need to use axios. That code was just using `fetch`

File: web/pages/product.jsx (hunk 3 above)
Comment (Sharookhan) [reply web/pages/product.jsx:13]: You mean copy the code/file?
> "server\services\product.service.js" is this the file?

File: web/pages/product.jsx (hunk 1 above)
Comment (Sharookhan) [reply web/pages/product.jsx:33]: Done

File: web/components/product/ReviewSection.jsx (hunk 2 above)
Comment (Sharookhan) [reply web/components/product/ReviewSection.jsx:3]: Done

File: web/pages/product.jsx (hunk 3 above)
Comment (georgevarghese185) [reply web/pages/product.jsx:13]: no I think it's `public/v1/J's/services/product-service.js`

**Review by georgevarghese185 — APPROVED** [review]
Looks Good!

**Pull request merged**